src/vm/program.ts

```diff
@@ -58,3 +59,22 @@ export type Program = z.infer<typeof Program>;
 export function parseProgram(program: string): Program {
   return Program.parse(JSON.parse(program));
 }
+
+const Cairo1Program = z.object({
+  prime: z.string(),
+  compiler_version: z.string(),
+  bytecode: z
+    .array(z.string())
+    .transform((value) => value.map((v) => new Felt(BigInt(v)))),
+  hints,
+  entrypoint: z.number(),
+  builtins: z.array(z.string()),
+});
+
+type Program2 = z.infer<typeof Cairo1Program>;
```

Comment:
I think that there is one Program type, and that possibly Cairo0 program is Program & { ... }

Reply:
We could make a base Zod object with all shared fields between Cairo Zero and Cairo compilation artifacts and then extend it for both specific fields. I'll propose a refactor